orchestration/dags/dependencies/firebase/sql/analytics/firebase_booking_origin.sql

```diff
@@ -0,0 +1,134 @@
+WITH firebase_bookings AS (
+  SELECT
+    user_id
+    , event_date AS booking_date
+    , event_timestamp AS booking_timestamp
+    , session_id AS booking_session_id
+    , offer_id
+    , booking_id
+    , platform
+    , search_id
+  FROM
+      `{{ bigquery_analytics_dataset }}.firebase_events` f_events
+  WHERE
+      event_name = 'BookingConfirmation'
+  -- Calculer sur les 4 derniers mois
+  AND event_date >= DATE('{{ add_days(ds, -120) }}')
+)
+
+, firebase_consult AS (
+  SELECT
+    user_id
+    , offer_id
+    , event_date AS consult_date
+    , event_timestamp AS consult_timestamp
+    , origin AS consult_origin
+    , reco_call_id
+    , module_id
+    , module_name
+    , entry_id
+  FROM `{{ bigquery_analytics_dataset }}.firebase_events`
+  WHERE event_name = 'ConsultOffer'
+  AND event_date >= DATE('{{ add_days(ds, -127) }}')
+)
+
+, bookings_origin_first_touch AS (
+  SELECT 
+    firebase_bookings.user_id
+    , booking_date
+    , booking_timestamp
+    , booking_session_id
+    , reco_call_id
+    , firebase_bookings.offer_id
+    , booking_id
+    , consult_date
+    , consult_timestamp
+    , consult_origin AS consult_origin_first_touch
+    , platform
+    , search_id
+    , module_id
+    , module_name
+    , entry_id
+  FROM firebase_bookings
+  LEFT JOIN firebase_consult
+  ON firebase_bookings.user_id = firebase_consult.user_id
+  AND firebase_bookings.offer_id = firebase_consult.offer_id
+  AND consult_date >= DATE_SUB(booking_date, INTERVAL 7 DAY)
```

Comment:
Peut être ajouter un AND consult_timestamp < booking_timestamp pour ne pas prendre les consultations survenues juste après la réservation